crates/ra_hir_ty/src/db.rs

```diff
@@ -111,7 +112,15 @@ pub trait HirDatabase: DefDatabase {
 }
 
 fn infer(db: &impl HirDatabase, def: DefWithBodyId) -> Arc<InferenceResult> {
-    let _p = profile("wait_infer");
+    let _p = profile("wait_infer").detail(|| match def {
+        DefWithBodyId::FunctionId(it) => db.function_data(it).name.to_string(),
```

Comment:
it would be nicer to do this without causing salsa deps, right?

Reply:
Yup, but this won't trigger unless profiling is enabled, so I think it's ok enough

Reply:
... but I don't know if there's even a way to do that

Reply:
Profiling is controlled by `RA_PROFILE` environment variable